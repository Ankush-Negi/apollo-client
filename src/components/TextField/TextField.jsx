import React from 'react';
import propTypes from 'prop-types';
import { Value, Error } from './style';

const TextField = (props) => {
  const {
    value, error, onChange, onBlur, readOnly,
  } = props;
  return (
    <>
      <Value type="text" value={value} readOnly={readOnly} onChange={onChange} onBlur={onBlur} />
      {error ? <Error>{error}</Error> : <br />}
    </>
  );
};

TextField.propTypes = {
  readOnly: propTypes.bool,
  onBlur: propTypes.func,
  value: propTypes.string,
  onChange: propTypes.func,
  error: propTypes.string,
};
TextField.defaultProps = {
  onBlur: undefined,
  value: undefined,
  error: '',
  onChange: undefined,
  readOnly: undefined,
};

export default TextField;
