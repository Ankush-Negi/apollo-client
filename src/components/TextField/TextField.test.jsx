import React from 'react';
import { mount } from 'enzyme';
import TextField from './TextField';

describe('Button Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<TextField />);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('render textbox', () => {
    const clicked = wrapper.find('TextField');
    expect(clicked.length).toBe(1);
  });
  it('render text inside', () => {
    const clicked = wrapper.find('TextField').value;
    expect(clicked).toEqual(undefined);
  });
});
