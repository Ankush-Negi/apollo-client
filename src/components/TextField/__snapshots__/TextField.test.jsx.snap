// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<TextField
  error=""
>
  <styled.input
    type="text"
  >
    <input
      className="sc-AxjAm imxerM"
      type="text"
    />
  </styled.input>
  <br />
</TextField>
`;
